engine/step/step.go

```diff
@@ -263,14 +289,14 @@ func Run(st *Step, baseConfig map[string]json.RawMessage, values *values.Values,
 			st.State = StateToRetry
 		default:
 			st.Output, st.Metadata, st.Tags, err = runner.Exec(st.Name, baseCfgRaw, config, ctx)
-			if baseOutput != nil {
-				if st.Output != nil {
+			if len(baseOutputs) > 0 && st.Output != nil {
```

Comment:
We could have some `baseOutputs` without having `output` from the actual runner. In that case, we should apply `baseOutputs` anyway